//============================================================================//
//                                                                            //
//                Copyright Â© 2015 - 2020 Subterranean Security               //
//                                                                            //
//  This source file is subject to the terms of the Mozilla Public License    //
//  version 2. You may not use this file except in compliance with the MPL    //
//  as published by the Mozilla Foundation at:                                //
//                                                                            //
//    https://mozilla.org/MPL/2.0                                             //
//                                                                            //
//=========================================================S A N D P O L I S==//
syntax = "proto3";

package soi;

option java_package = "com.sandpolis.core.soi";

/**
 * The SO_Build object contains information about the build process.
 */
message SO_Build {

    // The version string in a.b.c format.
    string version = 1;

    // The build number which uniquely identifies this build.
    int32 number = 2;

    // The build time
    int64 time = 3;

    // Whether this build is a debug or release build
    bool debug = 4;

    // The build platform's name
    string platform = 5;

    // The build platform's Java version
    string java_version = 6;

    // The build platform's Gradle version
    string gradle_version = 7;
}
